kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/manager/ResourceManager.java

```diff
@@ -118,9 +141,10 @@ private final <T extends HasMetadata> void createResource(boolean waitReady, T..
                 if (Objects.equals(resource.getKind(), KafkaTopic.RESOURCE_KIND)) {
                     continue;
                 }
+                assert type != null;
                 if (!waitResourceCondition(resource, ResourceCondition.readiness(type))) {
```

Comment:
I find the fact that waitResourceCondition deletes the resource if the resource isn't ready surprising.   Why's this useful behaviour?

Reply:
It is inherited from long time ago when the project started. Next step from this PR will be refactor ResourceManager and LogCollector. 

I'd appreciate if we keep the scope of this PR, as this is an old code that is not related to the change.

Reply:
This deletion is to avoid leaving leftovers in the test execution. We store all the resources installed by the manager and if the installation is not well done, the resource is not stored and won't be deleted at the end of the test run.

Reply:
Yeah, the point I am making is that waitResourceCondition could be used to wait for any condition.   Deleting if the condition doesn't reach true seems weird.  Happy for this to be refactored later.